packages/hydrogen/src/product/VariantSelector.ts

```diff
@@ -156,20 +186,48 @@ export const getSelectedProductOptions: GetSelectedProductOptions = (
   });
 
   return selectedOptions;
-};
+}
 
-type GetFirstAvailableVariant = (
+/**
+ * Returns the first availableForSale variant from a list of variants. If no variants are available, it returns `undefined`.
+ * @param variants an array of product variants or connection from the [Storefront API](https://shopify.dev/docs/api/storefront/latest/objects/Product#connection-product-variants)
+ *
+ * @returns the first available variant or `undefined`
+ *
+ * @example
+ * ```js
+ * // routes/products/$handle.tsx
+ * import {getFirstAvailableVariant} from '@shopify/hydrogen';
+ *
+ *
+ * export function loader({params}) {
+ *  const {product} = await storefront.query(PRODUCT_VARIANTS_QUERY, {
+ *    variables: {handle: params.handle}
+ *  });
+ *  const firstAvailableVariant = getFirstAvailableVariant(product.variants);
+ *  return json({product, firstAvailableVariant});
+ * }
+ *
+ *
+ * const PRODUCT_VARIANTS_QUERY = `#graphql
+ *  query ProductVariants($handle: String!) {
+ *    product(handle: $handle) {
+ *     variants(first: 250) {
+ *       nodes {
+ *        availableForSale
+ *        id
+ *       }
+ *     }
+ *   }
+ *  }
+ * ```
+ */
+export function getFirstAvailableVariant(
   variants:
     | PartialDeep<ProductVariantConnection>
     | Array<PartialDeep<ProductVariant>>,
-) => PartialDeep<ProductVariant> | undefined;
-
-export const getFirstAvailableVariant: GetFirstAvailableVariant = (
```

Comment:
Changed to `function` signature to ensure consistency with JSDocs hints

Reply:
Hmmm, I need to make sure the generated docs handle this ok.